### DISCUSSION TITLE
tracking trends in usage

### DISCUSSION THREAD

**Discussion by aae42** [KSXGitHub/parallel-disk-usage#256] [Q&A] [answered]
thanks so much for building pdu!

i was wondering what tracking changes to disk usage might look like

for example, imagine running pdu on a system hourly and dumping to json

to compare usage from one day to another, you compare the jsons with a semantic diffing tool

something like https://www.jsondiff.com/ , or https://github.com/trailofbits/graphtage (this part is actually surprisingly hard)

there are a few things that would be nice to make this workflow work better, i wish pdu had the option of ignoring or not listing files, only directories

having something like this built in to pdu would be astounding... 

anyone have any thoughts on this functionality? for now i can't find a better way than having a human just compare the outputs of pdu at two different times

**Comment by KSXGitHub** [comment, marked as answer]
> i wish pdu had the option of ignoring or not listing files, only directories

You may do something like `pdu --json-output | prune-files | pdu --json-input` (you are suppose to implement `prune-files`)

> i was wondering what tracking changes to disk usage might look like
> 
> for example, imagine running pdu on a system hourly and dumping to json
> 
> to compare usage from one day to another, you compare the jsons with a semantic diffing tool

I'm not sure if this is what you'd like, but here's how I would do it (you are supposed to implement `sort-pdu-output`):

```sh
# run this on a schedule
pdu --json-output | sort-pdu-output | pdu --json-input > pdu-$(date -u +%Y-%m-%d).txt
```

```sh
# compare them
diff -u pdu-2024-01-01.txt pdu-2024-02-01.txt
```
